src/main/groovy/com/morpheusdata/nutanix/prismelement/plugin/utils/NutanixPrismElementApiService.groovy

```diff
@@ -18,20 +18,1970 @@
+				if(taskDetail.success == true && taskDetail.results.progress_status) {
+					if(taskDetail.results.progress_status == 'Succeeded') {
+						rtn.success = true
+						rtn.results = taskDetail.results
+						pending = false
+					} else if(taskDetail.results.progress_status == 'Failure' || taskDetail.results.progress_status == 'Failed') {
+						rtn.error = true
+						rtn.results = taskDetail.results
+						rtn.success = true
+						pending = false
+					}
+				} else if(taskDetail.errorCode == 500) {
+					log.warn("task api call - 500 error (bug in nutanix)")
+					rtn.errorCode = taskDetail.errorCode
+					pending = false
+				}
+				attempts ++
+				if(attempts > 350)
+					pending = false
+			}
+		} catch(e) {
+			log.error("An Exception Has Occurred",e)
+		}
+		return rtn
+	}
+
+	static checkVmReady(HttpApiClient client, opts, vmId) {
+		def rtn = [success:false]
+		try {
+			if(vmId == null)
+				return rtn
+			def pending = true
+			def attempts = 0
+			while(pending) {
+				sleep(1000l * 10l)
+				def vmDetails = loadVirtualMachine(client, opts, vmId)
+				log.debug("vmDetails - ${vmId}: ${vmDetails}")
+				if(vmDetails.success == true && vmDetails.vmDetails?.powerState) {
+					rtn.success = true
+					rtn.results = vmDetails
+					pending = false
+				}
+				attempts ++
+				if(attempts > 150)
+					pending = false
+			}
+		} catch(e) {
+			log.error("An Exception Has Occurred",e)
+		}
+		return rtn
+	}
+
+	//v3
+	static buildHeaders(Map headers, String username, String password) {
+		headers = (headers ?: [:]) + ['Content-Type':'application/json;','Accept':'application/json']
+		if(username && password) {
+			def creds = "${username}:${password}"
+			headers['Authorization'] = "Basic ${creds.getBytes().encodeBase64().toString()}".toString()
+		}
+		return headers
+	}
+
+	//utils
+	static getNutanixApiUrl(zone) {
+		def apiUrl = zone.serviceUrl ?: zone.getConfigMap()?.apiUrl
+		if(apiUrl) {
+			if(apiUrl.startsWith('http')) {
+				URIBuilder uriBuilder = new URIBuilder("${apiUrl}")
+				uriBuilder.setPath('')
+				return uriBuilder.build().toString()
+			}
+			return 'https://' + apiUrl + ':9440'
+		}
+		throw new Exception('no nutanix api url specified')
+	}
+
+	static getNutanixUsername(zone) {
+		def rtn = zone.serviceUsername ?: zone.credentialData?.username ?: zone.getConfigProperty('username')
+		if(!rtn) {
+			throw new Exception('no nutanix username specified')
+		}
+		return rtn
+	}
+
+	static getNutanixPassword(zone) {
+		def rtn = zone.servicePassword ?: zone.credentialData?.password ?: zone.getConfigProperty('password')
+		if(!rtn) {
+			throw new Exception('no nutanix password specified')
+		}
+		return rtn
+	}
+
+	static getNutanixApiVersion(Cloud cloud) {
+		return cloud.serviceVersion ?: 'v1'
+	}
+
+	static getNutanixApiNumber(Cloud cloud) {
+		return getNutanixApiVersion(cloud)?.replace('v', '')?.toDouble() ?: 1.0
+	}
+
+	static withClient(opts=[:], Closure cl) {
```

Comment:
remove withClient, should be replaced with plugin core's HttpApiClient.

Reply:
Will do. It appeared to be used in one spot only and I wasn't sure if it was doing something special. I'll go ahead and convert it